### PR TITLE
WIP I731 deleted goals

### PR DESCRIPTION
**Pull request by Serhiy91**

### CODE REVIEW

File: app/imports/client/store/util/updateStore.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const updateStore = (store, { query, variables, getNewData }) => {
+  const goalsData = store.readQuery({ query, variables });
```
Comment (Dartv) [inline app/imports/client/store/util/updateStore.js]: If you make a reusable function, name variables accordingly.

File: app/imports/client/store/util/updateStore.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const updateStore = (store, { query, variables, getNewData }) => {
```
Comment (Dartv) [inline app/imports/client/store/util/updateStore.js]: It's not simply updating the redux store, it's updating the apollo store by query, move this file to client/apollo folder and name it differently (maybe updateStoreByQuery or updateQuery?)
It's not fp friendly either. The most consistent argument (store in this case) should come last. I think this should be better:
```js
export const updateQuery = curry((getNewData, { query, variables }, store) => ...
```

File: app/imports/ui/react/dashboard/components/DashboardGoals.js [hunk 3]
```diff
@@ -30,12 +34,17 @@ const DashboardGoals = ({
...
+    render={({ items }) => (
+      <Fragment>
+        {!!items.length && (
+          <GoalsChart
+            key={joinIds(items)}
+            goals={items}
+            {...{ onScatterTap }}
+          />
+        )}
+        <RemovedGoalsContainer organizationId={organizationId} />
```
Comment (Dartv) [inline app/imports/ui/react/dashboard/components/DashboardGoals.js]: It's already a convention in this project to pass variables with the same name as object, so why not follow it?
```js
<RemovedGoalsContainer {...{ organizationId }} />
```

File: app/imports/ui/react/goals/components/RemovedGoalItem.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getFormattedDate } from '../../../../share/helpers';
+
+const RemovedGoalItem = ({ goal }) => (
```
Comment (Dartv) [inline app/imports/ui/react/goals/components/RemovedGoalItem.js]: No need for the className there. Do not pass the whole object unless it's really needed. Just pass title, sequentialId, etc.. You can just spread them from the parent component or use flattenProp from recompose:
```js
<RemovedGoalItem {...goal} />
```
But remember to add onlyUpdateForKeys from recompose also.
Also, remember to pass ...props to reusable components so we can customize the component when needed (wrap with styled-components for example)

File: app/imports/ui/react/goals/components/RemovedGoals.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import RemovedGoalItem from './RemovedGoalItem';
+
+const RemovedGoals = ({ goals, totalCount }) => {
+  if (totalCount === 0) {
+    return null;
+  }
+  return (
+    <div>
+      <hr />
+      <h4 className="dashboard-stats-title">
```
Comment (Dartv) [inline app/imports/ui/react/goals/components/RemovedGoals.js]: no need for the className

File: app/imports/ui/react/goals/helpers.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { over, reject, where, equals, dec, compose, append, inc } from 'ramda';
+import { lenses } from 'plio-util';
+
+export const deleteGoalFromCache = (_id, data) => compose(
+  over(lenses.goals.totalCount, dec),
+  over(lenses.goals.goals, reject(where({ _id: equals(_id) }))),
+)(data);
+
+export const addGoalToCache = (goal, data) => compose(
```
Comment (Dartv) [inline app/imports/ui/react/goals/helpers.js]: I already have those in plio-util, just not committed. Guess I'll refactor it later.

File: app/imports/ui/react/goals/components/RemovedGoalItem.js [hunk 7]
```diff
@@ -1,22 +1,28 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { compose, onlyUpdateForKeys } from 'recompose';
 import { getFormattedDate } from '../../../../share/helpers';
 
-const RemovedGoalItem = ({ goal }) => (
+const enhance = compose(
+  onlyUpdateForKeys(['deletedAt', 'title', 'sequentialId']),
```
Comment (Dartv) [inline app/imports/ui/react/goals/components/RemovedGoalItem.js]: Why use `compose` there? Simply use `onlyUpdateForKeys` as enhancer

File: app/imports/ui/react/goals/containers/GoalAddModalContainer.js [hunk 8]
```diff
@@ -82,11 +81,10 @@ export default namedCompose('GoalAddModalContainer')(
           },
         },
         update: (proxy, { data: { createGoal: { goal } } }) => {
-          updateStore(proxy, {
+          updateQueryCache(data => addGoal(goal, data), {
```
Comment (Dartv) [inline app/imports/ui/react/goals/containers/GoalAddModalContainer.js]: your `addGoal` function should be curried. Then simply: `updateQueryCache(addGoal(goal))`

File: app/imports/ui/react/goals/containers/GoalEditModalContainer.js [hunk 9]
```diff
@@ -73,16 +72,14 @@ export default namedCompose('GoalEditModalContainer')(
         },
         update: (proxy, { data: { deleteGoal: { goal: removedGoal } } }) => {
           const variables = { organizationId };
-          updateStore(proxy, {
+          updateQueryCache(data => deleteGoal(goal._id, data), {
```
Comment (Dartv) [inline app/imports/ui/react/goals/containers/GoalEditModalContainer.js]: same as above

File: app/imports/ui/react/goals/components/RemovedGoalItem.js (hunk 7 above)
Comment (Serhiy91) [reply app/imports/ui/react/goals/components/RemovedGoalItem.js]: I have just forgot to remove compose. I added mapProps here in my next task

**Comment by Serhiy91** [comment]
@Dartv fixed

**Comment by Dartv** [comment]
@Serhiy91 prefix this PR with "WIP" and make other related PRs to this branch

**Pull request merged**